src/test/ui/rfc-2091-track-caller/std-panic-locations.rs

```diff
@@ -0,0 +1,37 @@
+// run-pass
+
+#![feature(option_expect_none, option_unwrap_none)]
+
+//! Test that panic locations for `#[track_caller]` functions in std have the correct
+//! location reported.
+
+fn main() {
+    // inspect the `PanicInfo` we receive to ensure the right file is the source
+    std::panic::set_hook(Box::new(|info| {
+        let actual = info.location().unwrap();
+        if actual.file() != file!() {
```

Comment:
```suggestion
        assert_eq!(actual.file(), file!(), ...);
```

Reply:
Unfortunately because this is invoked inside a panic hook, we'd lose error reporting going this route. I did it this way at first and it was hard to debug a missing track_caller annotation.

Reply:
It's nice to hear your explanation. I don't have the concern anymore.